apps/tailwind-components/components/input/Int.vue

```diff
@@ -0,0 +1,46 @@
+<template>
+  <InputString
+    :id="id"
+    :aria-describedby="describedBy"
+    :disabled="disabled"
+    :placeholder="placeholder"
+    :required="required"
+    v-model="modelValue"
+    @focus="$emit('focus')"
+    @blur="$emit('blur')"
+    @input="emit('update:modelValue', $event.target.value)"
+    @keypress="handleKeyValidity"
+  />
+</template>
+
+<script setup lang="ts">
+import type { IInputProps } from "../../types/types";
+import constants from "../../../molgenis-components/src/components/constants";
+import {
+  flipSign,
+  isNumericKey,
+} from "../../../molgenis-components/src/components/utils";
+
+const modelValue = defineModel<string | undefined>({ required: true });
+
+defineProps<
+  IInputProps & {
+    required?: boolean;
+  }
+>();
+
+const { CODE_MINUS, CODE_PERIOD } = constants;
+
+const emit = defineEmits(["focus", "blur", "update:modelValue"]);
```

Comment:
if you use modelValue you don't need to declare "update:modelValue" (or doesn't linter get that).

Reply:
Ah ok, other inputs are doing this so I thought I might be another weird nuxt thing.

Reply:
It is needed because it also does an emit in the script part.